extensions/panache/mongodb-panache/runtime/src/main/java/io/quarkus/mongodb/panache/transaction/interceptor/MongoTransaction.java

```diff
@@ -0,0 +1,63 @@
+package io.quarkus.mongodb.panache.transaction.interceptor;
+
+import javax.transaction.HeuristicMixedException;
+import javax.transaction.HeuristicRollbackException;
+import javax.transaction.RollbackException;
+import javax.transaction.Status;
+import javax.transaction.Synchronization;
+import javax.transaction.SystemException;
+import javax.transaction.Transaction;
+import javax.transaction.xa.XAResource;
+
+import com.mongodb.client.ClientSession;
+
+public class MongoTransaction implements Transaction {
+    private ClientSession clientSession;
+    private int status = Status.STATUS_ACTIVE;
+
+    public MongoTransaction(ClientSession clientSession) {
+        this.clientSession = clientSession;
+    }
+
+    public ClientSession getClientSession() {
+        return clientSession;
+    }
+
+    @Override
+    public void commit() throws RollbackException, HeuristicMixedException, HeuristicRollbackException, SecurityException,
+            IllegalStateException, SystemException {
+        clientSession.commitTransaction();
+        this.status = Status.STATUS_COMMITTED;
+    }
+
+    @Override
+    public boolean delistResource(XAResource xaRes, int flag) throws IllegalStateException, SystemException {
+        return false;
+    }
+
+    @Override
+    public boolean enlistResource(XAResource xaRes) throws RollbackException, IllegalStateException, SystemException {
+        return false;
+    }
+
+    @Override
+    public int getStatus() throws SystemException {
+        return status;
+    }
+
+    @Override
+    public void registerSynchronization(Synchronization sync) throws RollbackException, IllegalStateException, SystemException {
+
+    }
+
+    @Override
+    public void rollback() throws IllegalStateException, SystemException {
+        clientSession.abortTransaction();
+        this.status = Status.STATUS_ROLLEDBACK;
+    }
+
+    @Override
+    public void setRollbackOnly() throws IllegalStateException, SystemException {
```

Comment:
You need to support this, it's pretty useful.

Reply:
And easy to implement.

Reply:
Yes, done ;)